### PR TITLE
[SPARK-26170][SS] Add missing metrics in FlatMapGroupsWithState

### PR DESCRIPTION
**Pull request by HeartSaVioR**
## What changes were proposed in this pull request?

This patch addresses measuring possible metrics in StateStoreWriter to FlatMapGroupsWithStateExec. Please note that some metrics like time to remove elements are not addressed because they are coupled with state function.

## How was this patch tested?

Manually tested with https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/sql/streaming/StructuredSessionization.scala.

Snapshots below:

screen shot 2018-11-26 at 4 13 40 pm
screen shot 2018-11-26 at 4 13 54 pm

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
retest this, please

**Comment by HeartSaVioR** [comment]
cc. @tdas @zsxwing

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by HeartSaVioR** [comment]
Kindly ping to @tdas @zsxwing again. This is a simple patch.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FlatMapGroupsWithStateExec.scala [hunk 1]
```diff
@@ -111,28 +111,41 @@ case class FlatMapGroupsWithStateExec(
       indexOrdinal = None,
       sqlContext.sessionState,
       Some(sqlContext.streams.stateStoreCoordinator)) { case (store, iter) =>
+        val numOutputRows = longMetric("numOutputRows")
```
Comment (gaborgsomogyi) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FlatMapGroupsWithStateExec.scala]: Just wondering what's the difference between this and the metric with the same name in `InputProcessor`? Don't they collide somehow?

Comment (gaborgsomogyi) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FlatMapGroupsWithStateExec.scala]: ping @HeartSaVioR

Comment (HeartSaVioR) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FlatMapGroupsWithStateExec.scala]: Sorry missed this. I accidentally marked this as resolved at mobile so rolled back. Will get back when I got time.

Comment (HeartSaVioR) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FlatMapGroupsWithStateExec.scala]: You're right I missed the metrics in InputProcessor. We can remove `numOutputRows` here. Thanks for finding missing spot!

Comment (HeartSaVioR) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FlatMapGroupsWithStateExec.scala]: Addressed.

**Review by gaborgsomogyi — APPROVED** [review]

**Comment by HeartSaVioR** [comment]
Kindly reminder.

**Comment by HeartSaVioR** [comment]
Ping @tdas again, the patch is pretty simple to be reviewed within 10 mins.

**Review by jose-torres — APPROVED** [review]
LGTM. Merging to master.

**Comment by dongjoon-hyun** [comment]
Wow, congratulation, @jose-torres !

**Comment by HeartSaVioR** [comment]
Thanks @jose-torres for reviewing and merging! And you're becoming a committer :) Congrats!

**Comment by jose-torres** [comment]
Thanks!

**Pull request closed without merging**